CRM/Contribute/BAO/Contribution.php

```diff
@@ -5457,4 +5457,28 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Calculate tax for individual line items.
+   *
+   * @param array $lineItem
+   *   an array of lineitems
+   *
+   * @param int $contributionId
+   *   the id of the contribution
+   *
+   * @return array
```

Comment:
Thanks for the input @JoeMurray - I hadn't gone down to the code level on these PRs yet, because I wanted to start from an understanding of what they are trying to fix